be/src/vec/functions/function_java_udf.cpp

```diff
@@ -0,0 +1,210 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "vec/functions/function_java_udf.h"
+
+#include <fmt/format.h>
+
+#include <memory>
+#include <sstream>
+
+#include "runtime/exec_env.h"
+#include "runtime/user_function_cache.h"
+
+#include "util/jni-util.h"
+#include "vec/columns/column_vector.h"
+#include "vec/core/block.h"
+#include "vec/data_types/data_type_bitmap.h"
+#include "vec/data_types/data_type_date.h"
+#include "vec/data_types/data_type_date_time.h"
+#include "vec/data_types/data_type_decimal.h"
+#include "vec/data_types/data_type_nullable.h"
+#include "vec/data_types/data_type_number.h"
+#include "vec/data_types/data_type_string.h"
+
+const char* EXECUTOR_CLASS = "org/apache/doris/udf/UdfExecutor";
+const char* EXECUTOR_CTOR_SIGNATURE ="([B)V";
+const char* EXECUTOR_EVALUATE_SIGNATURE = "()V";
+const char* EXECUTOR_CLOSE_SIGNATURE = "()V";
+
+namespace doris::vectorized {
+JavaFunctionCall::JavaFunctionCall(const TFunction& fn,
+                     const DataTypes& argument_types, const DataTypePtr& return_type)
+        : fn_(fn),
+          _argument_types(argument_types),
+          _return_type(return_type) {}
+
+Status JavaFunctionCall::prepare(FunctionContext* context, FunctionContext::FunctionStateScope scope) {
+    DCHECK(executor_cl_ == NULL) << "Init() already called!";
+    JNIEnv* env = JniUtil::GetJNIEnv();
+    if (env == NULL) return Status::InternalError("Failed to get/create JVM");
+    RETURN_IF_ERROR(JniUtil::GetGlobalClassRef(env, EXECUTOR_CLASS, &executor_cl_));
+    executor_ctor_id_ = env->GetMethodID(
+            executor_cl_, "<init>", EXECUTOR_CTOR_SIGNATURE);
+    RETURN_ERROR_IF_EXC(env);
+    executor_evaluate_id_ = env->GetMethodID(
+            executor_cl_, "evaluate", EXECUTOR_EVALUATE_SIGNATURE);
+    RETURN_ERROR_IF_EXC(env);
+    executor_close_id_ = env->GetMethodID(
+            executor_cl_, "close", EXECUTOR_CLOSE_SIGNATURE);
+    RETURN_ERROR_IF_EXC(env);
+
+    int64_t input_values_buffer_ptr = (int64_t) new int64_t[_argument_types.size()];
```

Comment:
the mem only user in `JniContext`, why not alloc and release in `JniContext` class ?

Reply:
Sure, this is a better choice.